### ISSUE TITLE
Adding analytics fails with "Parameters: [auth...IdentityPoolId] do not exist in the template"

### ISSUE THREAD

**Issue by cunneen** [aws-amplify/amplify-cli#11247]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.x

### Amplify CLI Version

10.3.1

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No changes at all, I only used the command-line

### Amplify Categories

auth, analytics

### Amplify Commands

add, push

### Describe the bug

Possibly related to #11148 and #9525

Push fails after adding analytics, seemingly when trying to reference a newly-defined Identity Pool ID.

```
2022-10-24 15:44:22 UTC+0800 | authuserPoolGroups | UPDATE_FAILED | Parameters: [authDummyAuthIdentityPoolId] do not exist in the template
-- | -- | -- | --
```

### Expected behavior

Push succeeds

### Reproduction steps

```zsh
npx create-react-app amplify-pinpoint-dummy
cd amplify-pinpoint-dummy
amplify init -y
amplify add auth # see below for full transcript
amplify push
amplify add analytics # see below for full transcript
amplify push
```

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Project Identifier

cf48e1cc842434273464acf75094c763

### Log output

[amplify add auth interactive CLI transcript]
<details>

```
Using service: Cognito, provided by: awscloudformation
 
 The current configured provider is Amazon Cognito. 
 
 Do you want to use the default authentication and security configuration? Manual configuration
 Select the authentication/authorization services that you want to use: User Sign-Up & Sign-In only
 (Best used with a cloud API only)
 Provide a friendly name for your resource that will be used to label this category in the project:
 DummyAuth
 Provide a name for your user pool: DummyUserPoolDeleteMe
 Warning: you will not be able to edit these selections. 
 How do you want users to be able to sign in? Email
 Do you want to add User Pool Groups? Yes
? Provide a name for your user pool group: Admins
? Do you want to add another User Pool Group Yes
? Provide a name for your user pool group: Users
? Do you want to add another User Pool Group No
✔ Sort the user pool groups in order of preference · Users, Admins
 Do you want to add an admin queries API? Yes
? Do you want to restrict access to the admin queries API to a specific Group Yes
? Select the group to restrict access with: Admins
 Multifactor authentication (MFA) user login options: OPTIONAL (Individual users can use MFA)
 For user login, select the MFA types: Time-Based One-Time Password (TOTP)
 Specify an SMS authentication message: Your authentication code is {####}
 Email based user registration/forgot password: Enabled (Requires per-user email entry at registrat
ion)
 Specify an email verification subject: Your verification code
 Specify an email verification message: Your verification code is {####}
 Do you want to override the default password policy for this User Pool? No
 Warning: you will not be able to edit these selections. 
 What attributes are required for signing up? Email, Name
 Specify the app's refresh token expiration period (in days): 30
 Do you want to specify the user attributes this app can read and write? Yes
 Specify read attributes: Email, Name, Phone Number
 Specify write attributes: Phone Number
 Do you want to enable any of the following capabilities? Email Verification Link with Redirect, Ad
d User to Group
 Do you want to use an OAuth flow? No
? Do you want to configure Lambda Triggers for Cognito? Yes
? Which triggers do you want to enable for Cognito Custom Message, Post Confirmation, Pre Token Gen
eration
? What functionality do you want to use for Custom Message Send Account Confirmation Link w/ Redire
ct
? What functionality do you want to use for Post Confirmation Add User To Group
? What functionality do you want to use for Pre Token Generation Create your own module
✔ Enter the URL that your users will be redirected to upon account confirmation: · 
✔ Enter the subject for your custom account confirmation email: · 
✔ Enter the body text for your custom account confirmation email (this will appear before the link URL): · 

Successfully added resource DummyAuthCustomMessage locally.

Next steps:
Check out sample function code generated in <project-dir>/amplify/backend/function/DummyAuthCustomMessage/src
"amplify function build" builds all of your functions currently in the project
"amplify mock function <functionName>" runs your function locally
To access AWS resources outside of this Amplify app, edit the /Users/mikecunneen/Development/bestperformance/amplify-pinpoint-dummy/amplify/backend/function/DummyAuthCustomMessage/custom-policies.json
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all of your local backend and front-end resources (if you added hosting category) and provisions them in the cloud
Successfully added the Lambda function locally
? Do you want to edit your verification-link function now? Yes
Edit the file in your editor: /Users/mikecunneen/Development/bestperformance/amplify-pinpoint-dummy/amplify/backend/function/DummyAuthCustomMessage/src/verification-link.js
? Press enter to continue 
✔ Enter the name of the group to which users will be added. · Users
Successfully added resource DummyAuthPostConfirmation locally.

Next steps:
Check out sample function code generated in <project-dir>/amplify/backend/function/DummyAuthPostConfirmation/src
"amplify function build" builds all of your functions currently in the project
"amplify mock function <functionName>" runs your function locally
To access AWS resources outside of this Amplify app, edit the /Users/mikecunneen/Development/bestperformance/amplify-pinpoint-dummy/amplify/backend/function/DummyAuthPostConfirmation/custom-policies.json
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all of your local backend and front-end resources (if you added hosting category) and provisions them in the cloud
Successfully added the Lambda function locally
? Do you want to edit your add-to-group function now? Yes
Edit the file in your editor: /Users/mikecunneen/Development/bestperformance/amplify-pinpoint-dummy/amplify/backend/function/DummyAuthPostConfirmation/src/add-to-group.js
? Press enter to continue 
Successfully added resource DummyAuthPreTokenGeneration locally.

Next steps:
Check out sample function code generated in <project-dir>/amplify/backend/function/DummyAuthPreTokenGeneration/src
"amplify function build" builds all of your functions currently in the project
"amplify mock function <functionName>" runs your function locally
To access AWS resources outside of this Amplify app, edit the /Users/mikecunneen/Development/bestperformance/amplify-pinpoint-dummy/amplify/backend/function/DummyAuthPreTokenGeneration/custom-policies.json
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all of your local backend and front-end resources (if you added hosting category) and provisions them in the cloud
Successfully added the Lambda function locally
? Do you want to edit your custom function now? Yes
Edit the file in your editor: /Users/mikecunneen/Development/bestperformance/amplify-pinpoint-dummy/amplify/backend/function/DummyAuthPreTokenGeneration/src/custom.js
? Press enter to continue 
✅ Successfully added AdminQueriesbcad9d5a function locally
✅ Successfully added AdminQueries API locally
✅ Successfully added auth resource DummyAuth locally

✅ Some next steps:
"amplify push" will build all your local backend resources and provision it in the cloud
"amplify publish" will build all your local backend and frontend resources (if you have hosting category added) and provision it in the cloud
```

</details>

[amplify add analytics transcript]
<details>

```
? Select an Analytics provider Amazon Pinpoint
? Provide your pinpoint resource name: amplifypinpointdummy
⚠️ Current auth configuration is: userPoolOnly, but identityPoolOnly was required.
Auth configuration is required to allow unauthenticated users, but it is not configured properly.
⚠️ Adding analytics would add the Auth category to the project if not already added.
? Apps need authorization to send analytics events. Do you want to allow guests and unauthenticated
 users to send analytics events? (we recommend you allow this when getting started) No
⚠️ Authorize only authenticated users to send analytics events. Use "amplify update auth" to modify this behavior.
✅ Successfully updated auth resource locally.
✅ Successfully added resource amplifypinpointdummy locally

✅ Some next steps:
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all your local backend and front-end resources (if you have hosting category added) and provisions them in the cloud
```
</details>

**Comment by ykethan** [comment]
Hey @cunneen, thank you for reaching out. I was able to reproduce the issue using the provided steps in `log output`. 

observed the following errors in cloudformation.
`
  | Parameters: [authauth11247fe5ce64aIdentityPoolId] do not exist in the template
`

`
The following resource(s) failed to create: [UpdateRolesWithIDPFunction, analyticsauth11247]. The following resource(s) failed to update: [functionAdminQueries7441062e, authuserPoolGroups, AuthTriggerCustomLambdaStack].
`

Marking this as a bug.

**Comment by ykethan** [comment]
Hey @cunneen, could you test this in the latest Amplify CLI v10.4.0 and let us know if this mitigates the issue.

**Comment by edwardfoyle** [comment]
Additionally, could you check if the entry for the auth userPoolGroups in `amplify/backend/backend-config.json` contains the following in the `dependsOn` block:

![Screen Shot 2022-10-27 at 17 08 15](https://user-images.githubusercontent.com/19419978/198420094-cf9a5b8e-c647-4e3e-9550-b1e0cd35026d.png)

**Comment by danielleadams** [comment]
Removed `pending-response` label, but have talked to @ykethan and he has observed this issue happens intermittently.

**Comment by cunneen** [comment]
Thanks for the update @danielleadams .

**Comment by lazpavel** [comment]
Hi @cunneen, this issue is fixed in the latest Amplify CLI (> v10.8.1).
Tested above steps and push completed successfully.
